### PR TITLE
Add toasts for sign-up validation and implement tests

### PR DESCRIPTION
**Pull request by ethanboren**
**Description:**

This pull request enhances the sign-up process by adding toast messages to provide immediate feedback to users when form validation rules are not met. It also includes unit tests to ensure the validation logic functions correctly.

### Changes Made:
**Added Toast Messages:**
- Display a toast when the form is incomplete.
- Display a toast when passwords do not match.
- Display a toast when the password is less than 6 characters.

**Motivation:**
- Improving user experience by providing clear and immediate feedback during the sign-up process. The added tests ensure that any future changes to the validation logic do not break existing functionality.

**Testing:**
- Performed manual testing to verify that the toast messages appear as expected.

**Notes:**
- The refactored code enhances maintainability and ease of testing.
- Feedback from the toast messages should help reduce user frustration and improve the overall sign-up experience.

### CODE REVIEW

**Review by hjamaly — APPROVED** [review]
LGTM ! Toast were correctly added ! Good job ! However Testing is irrelevant since you're not mocking the auth view model !

File: app/src/main/java/com/android/solvit/shared/ui/authentication/SignUpScreen.kt
Comment (achillet3) [inline app/src/main/java/com/android/solvit/shared/ui/authentication/SignUpScreen.kt:1]: Good implementation of the additional toasts!

**Review by achillet3 — CHANGES_REQUESTED** [review]
## Overall impression
- Good implementation of toasts to improve user experience by providing immediate feedback. 
- Good refactoring of code to help maintainability

## Improvements
- Testing of the screen should be done by mocking the authentification view model to be sure the UI interacts correctly with the view model and the sign up button behaves correctly.

**Review by achillet3 — APPROVED** [review]
Took out the incorrect tests, and passed all tests and coverage minimum. LGTM!

**Pull request merged**